source/mesh_refinement/isotherms.cc

```diff
@@ -0,0 +1,373 @@
+                              if (cell->level() <  isotherms_levels[isotherm_line].first)
+                                {
+                                  refine = true;
+                                }
+
+                              // If the current refinement level is larger or equal to the maximum refinement level, any refinement flag should be cleared.
+                              if (cell->level() >= isotherms_levels[isotherm_line].second)
+                                {
+                                  clear_refine = true;
+                                }
+                              // If the current refinement level is larger then the maximum level, a coarsening flag should be placed, and we don't have to look any further.
+                              if (cell->level() >  isotherms_levels[isotherm_line].second)
+                                {
+                                  coarsen = true;
+                                }
+
+                            }
+                        }
+                    }
+                }
+
+              // if both coarsen and refine are true, give preference to refinement
+              if (coarsen == true && refine == true)
+                {
+                  coarsen = false;
+                  clear_refine = false;
+                }
+
+              // Perform the actual placement of the coarsening and refinement flags
+              // We want to make sure that the refiment never goes below the minimum
+              // or above the maximum, so we first check/set the coarsen/refine flag,
+              // and then check/set the clear coarsen/refine flag.
+              if (coarsen == true)
+                {
+                  cell->set_coarsen_flag ();
+                }
+              if (clear_coarsen == true)
+                {
+                  cell->clear_coarsen_flag ();
+                }
+              if (refine == true)
+                {
+                  cell->set_refine_flag ();
+                }
+              if (clear_refine == true)
+                {
+                  cell->clear_refine_flag ();
+                }
+            }
+        }
+    }
+
+    template <int dim>
+    void
+    Isotherms<dim>::
+    declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Mesh refinement");
+      {
+        prm.enter_subsection("Isotherms");
+        {
+          prm.declare_entry ("Exclude composition", "-1",
+                             Patterns::Double (),
+                             "This option allows to exclude one composition from being condidered by this plugin. "
+                             "This can for example be used to exclude the composition representing a sticky air, which "
+                             "is as cold as the surface, but you do not want to be refined."
```

Comment:
You should say somewhere that this variable is the *index* of one compositional field that allows to disable the plugin, *if the compositional field has a value less than 0.5*.